app/code/community/Bolt/Boltpay/controllers/ApiController.php

```diff
@@ -195,10 +195,43 @@ public function hookAction()
                     ->setHeader("Retry-After", "86400")
                     ->setBody(json_encode(array('status' => 'failure', 'error' => array('code' => 6009, 'message' => $boltHelperBase->__('The order is on-hold and requires manual update before this hook is accepted') ))));
             } else {
-                // An invalid transition is treated as a late queue event and hence will be ignored
-                //Mage::log($errorMessage, null, 'bolt.log');
-                //Mage::log("Late queue event. Returning as OK", null, 'bolt.log');
-                $this->getResponse()->setHttpResponseCode(200);
+                /////////////////////////////////////////////////////////////
```

Comment:
I think we can benefit from having variables with proper naming. can you do something like:

```
 if ($boltPayInvalidTransitionException->getOldStatus() == Bolt_Boltpay_Model_Payment::TRANSACTION_ON_HOLD) {
    $this->getResponse()->setHttpResponseCode(503)
        ->setHeader("Retry-After", "86400")
        ->setBody(json_encode(array('status' => 'failure', 'error' => array('code' => 6009, 'message' => $boltHelperBase->__('The order is on-hold and requires manual update before this hook is accepted') ))));
    return;
} 

$isCreditOrCapture = in_array($hookType, array('credit', 'capture');
$isSameTransition = $newTransactionStatus === $prevTransactionStatus;
$rejectsCanelledOrder = $prevTransactionStatus === Bolt_Boltpay_Model_Payment::TRANSACTION_CANCELLED)
                && in_array($hookType, array('void','rejected_reversible','rejected_irreversible');
$authForCompletedOrder = $prevTransactionStatus === Bolt_Boltpay_Model_Payment::TRANSACTION_COMPLETED)
                && in_array($hookType, array('auth','payment');

$isDuplicateTransition = $isDuplicateTransition || $rejectsCanelledOrder || $authForCompletedOrder;

if (!in_array($hookType, array('credit', 'capture') && isDuplicateTransition) {
    // return 200
} else {
    // return 422
}
```

Reply:
No problem. 

 I can agree with this style over making separate methods that spread localized logic over several methods and files because it allows for both an easy read and also the details of implementation immediately known while reducing the need for comments.  The comments are in the code and the software engineer is more directly confronted to maintain it.

Thanks!